destination/destination.go

```diff
@@ -0,0 +1,141 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+
+	cf "github.com/meroxa/conduit-connector-amazon-sqs/config"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type Destination struct {
+	sdk.UnimplementedDestination
+	config   cf.Config
+	svc      *sqs.Client
+	queueURL *string
+}
+
+func NewDestination() sdk.Destination {
+	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
+}
+
+func (d *Destination) Parameters() map[string]sdk.Parameter {
+	return map[string]sdk.Parameter{
+		cf.ConfigKeyAWSAccessKeyID: {
+			Default: "",
+			Validations: []sdk.Validation{
+				sdk.ValidationRequired{},
+			},
+			Description: "AWS Access Key ID.",
+		},
+		cf.ConfigKeyAWSSecretAccessKey: {
+			Default: "",
+			Validations: []sdk.Validation{
+				sdk.ValidationRequired{},
+			},
+			Description: "AWS Secret Access Key.",
+		},
+		cf.ConfigKeyAWSToken: {
+			Default:     "",
+			Description: "AWS Access Token (optional).",
+		},
+		cf.ConfigKeyAWSQueue: {
+			Default: "",
+			Validations: []sdk.Validation{
+				sdk.ValidationRequired{},
+			},
+			Description: "AWS SQS Queue Name.",
+		},
+	}
+}
+
+func (d *Destination) Configure(ctx context.Context, cfg map[string]string) error {
+	sdk.Logger(ctx).Debug().Msg("Configuring Destination Connector.")
+	parsedCfg, err := cf.ParseConfig(cfg)
+	if err != nil {
+		return err
+	}
+	d.config = parsedCfg
+
+	return nil
+}
+
+func (d *Destination) Open(ctx context.Context) error {
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithCredentialsProvider(
+			credentials.NewStaticCredentialsProvider(
+				d.config.AWSAccessKeyID,
+				d.config.AWSSecretAccessKey,
+				d.config.AWSToken)),
+	)
+	if err != nil {
+		return err
+	}
+	// Create a SQS client from just a session.
+	d.svc = sqs.NewFromConfig(cfg)
+
+	queueInput := &sqs.GetQueueUrlInput{
+		QueueName: &d.config.AWSQueue,
+	}
+
+	// Get URL of queue
+	urlResult, err := d.svc.GetQueueUrl(ctx, queueInput)
+	if err != nil {
+		return err
+	}
+
+	d.queueURL = urlResult.QueueUrl
+
+	return nil
+}
+
+func (d *Destination) Write(ctx context.Context, records []sdk.Record) (int, error) {
+	for i, record := range records {
+		messageBody := string(record.Payload.After.Bytes())
+
+		messageAttributes := map[string]types.MessageAttributeValue{}
+		for key, value := range record.Metadata {
+			messageAttributes[key] = types.MessageAttributeValue{
+				DataType:    aws.String("String"),
+				StringValue: aws.String(value),
+			}
+		}
+		// construct record to send to destination
+		sendMessageInput := &sqs.SendMessageInput{
+			DelaySeconds:      10,
```

Comment:
I guess there is no need, one of the default examples I saw added delay for some reason, but its not necessary

Reply:
I'm not that familiar with SQS to know how important it is. That said, it would be pretty simple to create a config option and let the user configure this if they want, and let the default be 0, I think we should do it.